### PR TITLE
Implement changes for SAPP-2130

### PR DESCRIPTION
**Pull request by singularo**
Knocked it up quickly, needs testing.

### CODE REVIEW

**Comment by behzad1q** [comment]
it should be working but not 
image

File: scaffold/required/monolog.services.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+parameters:
```
Comment (dpi) [inline scaffold/required/monolog.services.yml:1]: This file should be moved to `scaffold/required/web/sites/default/monolog.services.yml`.

You can get inspiration from the docker files in `scafffold/optional/docker/`

**Review by dpi — CHANGES_REQUESTED** [review]
path structures need tweaking

**Comment by dpi** [comment]
Should the remaining steps for how to activate monolog be added to Monolog? ie. composer.json, core.extension.yml entry, ...

**Comment by singularo** [comment]
This seems to work, but having slept on it, I'm not sure we should be merging this, or even using monolog at all.

1. Monolog is cool and has lots of features, we're not using any of them, or planning to use any that i'm aware of.
2. We could integrate sending messages directly to splunk, but our experience doing things that way isn't good, so I don't think thats likely, especially given openshift will be logging to spunk already.
3. So then these scaffold changes add no new features?
4. Putting this in the scaffold makes a hard dependency in all projects that use it, and uninstalling monolog then breaks things:
```
$ drush cr
In DefinitionErrorExceptionPass.php line 54:
  You have requested a non-existent parameter "monolog.level.debug".
```
5. The settings.app.php could be used with a configmap to achieve the same result if individual sites require it.
6. monolog adds 10x the code for a bunch of features we're not using (and yes I know, not maintaining either).
144K	web/modules/contrib/monolog
724K	vendor/monolog/
88K	web/modules/contrib/log_stdout/

Unless we're logging to something other than stdout, why are we even switching?

**Comment by pingers** [comment]
A couple of reasons I guess:
1. Monolog is the standard for php logging.
2. We can use a template for the log format - more parser friendly for splunk... one day.

**Comment by dpi** [comment]
Ultimately it doesnt *need* to be in scaffold.

I think we just want to get off https://www.drupal.org/project/log_stdout onto something more standard.

**Comment by singularo** [comment]
I'll leave it to @pingers to decide ;-P

**Comment by pingers** [comment]
Need to check that we can still log to stdout + dblog. E.g. for multi-line log entries.

**Comment by singularo** [comment]
With monolog enabled and configured as above, enabling dblog doesn't actually get any dblogs.
Changing the monolog.services.yml to have:
```
parameters:
  monolog.channel_handlers:
    default: ['drupal.dblog']
    php: ['drupal.db_log']
```

Does then log to dblog.

But, this requires changing the scaffold file in this commit, which would need to be overwritten with a configmap.

I feel that monolog is setup/configured at a lower level than works well in the OpenShift environment, so I'm going to suggest we not merge this, and in fact do move back to log_stdout for our use case, which is just simple stdout logging.

**Comment by singularo** [comment]
Closing this

**Pull request closed without merging**